File: sdks/node/src/api/core/cyan_res.ts
interface CyanGlobRes {
  glob: string;
  exclude: string[];
  type: string;
}

interface CyanPluginRes {
  name: string;
  config: unknown;
}

interface CyanProcessorRes {
  name: string;
  config: unknown;
  files: CyanGlobRes[];
}

interface CyanRes {
  processors: CyanProcessorRes[];
  plugins: CyanPluginRes[];
}

// export all
export type { CyanRes, CyanProcessorRes, CyanPluginRes, CyanGlobRes };

File: sdks/dotnet/sulfone-helium/Api/Core/CyanRes.cs
using System.Dynamic;

namespace sulfone_helium.Api.Core;

public readonly struct CyanGlobRes
{
    public string? Root { get; init; }
    public string Glob { get; init; }
    public string[] Exclude { get; init; }
    public string Type { get; init; }
}

public readonly struct CyanPluginRes
{
    public string Name { get; init; }
    public ExpandoObject Config { get; init; }
}

public readonly struct CyanProcessorRes
{
    public string Name { get; init; }
    public ExpandoObject Config { get; init; }

    public CyanGlobRes[] Files { get; init; }
}

public readonly struct CyanRes
{
    public CyanProcessorRes[] Processors { get; init; }
    public CyanPluginRes[] Plugins { get; init; }
}
